### PR TITLE
[Mailing] add celery in project

### PR DESCRIPTION
**Pull request by paivazov**
Adds celery service in docker-compose. It's ready to receive tasks.

**References:**
- #16

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,5 +20,23 @@ services:
...
+    command: celery -A celery_app worker -l INFO
+    env_file:
+      - .env
+    environment:
+        - BROKER_URL=redis://redis/0
+        - RESULT_BACKEND=redis://redis/1
+    depends_on:
+        - api_server
+        - postgres_server
+        - redis
```
Comment (BorodaUA) [inline docker-compose.yml]: Do we need a docker service for such thing as running celery inside our api_server container? maybe we should put celery run command in enrypoint.sh?

File: docker-compose.yml [hunk 2]
```diff
@@ -20,5 +20,23 @@ services:
...
+  redis:
+      image: redis
+      ports:
+        - "6379:6379"
```
Comment (BorodaUA) [inline docker-compose.yml]: lets use .env file variables for variables like that.

File: api_server/celery_app.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from celery import Celery
+
+app = Celery(
+    "retraining",
+    backend="redis://redis/1",
+    broker="redis://redis/0",
```
Comment (BorodaUA) [inline api_server/celery_app.py]: Maybe it is better to put those in some kind of constants, and store it in single place?

File: docker-compose.yml [hunk 4]
```diff
@@ -20,5 +20,23 @@ services:
...
       - postgres_server
+  celery:
+    restart: always
+    build: ./api_server
+    command: celery -A celery_app worker -l INFO
+    env_file:
+      - .env
+    environment:
+        - BROKER_URL=redis://redis/0
+        - RESULT_BACKEND=redis://redis/1
```
Comment (BorodaUA) [inline docker-compose.yml]: lets use .env file variables for variables like that.

**Review by MaksymShvets — APPROVED** [review]

**Review by evroandrew — APPROVED** [review]

**Review by BorodaUA — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (paivazov) [reply docker-compose.yml]: Celery - it's separate docker container and it's independent of our api. I think calculations of celery app should take place in a separate container

File: docker-compose.yml (hunk 2 above)
Comment (paivazov) [reply docker-compose.yml]: Okay, If it has a reason to use in this case

File: api_server/celery_app.py (hunk 3 above)
Comment (paivazov) [reply api_server/celery_app.py]: Yeah, sure. Sorry, my mistake, I'll change it

File: docker-compose.yml (hunk 4 above)
Comment (paivazov) [reply docker-compose.yml]: Yes, now those variables are being pulled from `.env` file

File: api_server/celery_app.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+from celery import Celery
+
+app = Celery(
+    "retraining",
+    backend=os.environ.get('RESULT_BACKEND'),
+    broker=os.environ.get('BROKER_URL'),
+)
+
+app.autodiscover_tasks()
```
Comment (BorodaUA) [inline api_server/celery_app.py:11]: Will we have celery jobs? If so, will they be stored in project root? as celery_jobs.py? Maybe we need to create a module for that stuff? Could a module solve that? (not named celery \ __init__.py obviously)

**Review by BorodaUA — CHANGES_REQUESTED** [review]

**Review by BorodaUA — APPROVED** [review]
Done.

**Pull request merged**